nucliadb_node/src/bin/writer.rs

```diff
@@ -98,6 +95,9 @@ async fn main() -> NodeResult<()> {
     let (shutdown_notifier, shutdown_notified) = get_shutdown_notifier();
     let shard_cache = Arc::new(ShardWriterCache::new(settings.clone()));
 
+    lifecycle::initialize_writer(settings.clone())?;
+    lifecycle::initialize_merger(Arc::clone(&shard_cache), settings.clone())?;
+
```

Comment:
I was expecting to see a hook at the end of the writer (after `wait_for_sigkill`) to cleanly shutdown the merger. Maybe send a message that sets a boolean that makes the thread exit as soon as it finished the current job.

Not critical, since an unfinished merge will eventually be GC'ed, but still, it's good practice to have clean shutdowns.

Reply:
I added a `Mutex<bool>` in the `MergeScheduler` and stop function. Do you think is okay?

Reply:
Yes, `Mutex<bool>` works. An alternative would be to use `AtomicBool`. Both should work well for this.

I left a comment in the scheduler because I see `shutdown` being set, but I don't see it being read to exit the loop.